vllm/executor/ray_gpu_executor.py

```diff
@@ -257,19 +261,31 @@ def execute_model(self,
                       blocks_to_swap_in: Dict[int, int],
                       blocks_to_swap_out: Dict[int, int],
                       blocks_to_copy: Dict[int, List[int]]) -> SamplerOutput:
-        all_outputs = self._run_workers(
-            "execute_model",
-            driver_kwargs={
-                "seq_group_metadata_list": seq_group_metadata_list,
-                "blocks_to_swap_in": blocks_to_swap_in,
-                "blocks_to_swap_out": blocks_to_swap_out,
-                "blocks_to_copy": blocks_to_copy,
-            },
-            use_ray_compiled_dag=USE_RAY_COMPILED_DAG)
+        if self.parallel_worker_tasks is None:
+            # Start model execution loop running in the parallel workers
+            parallel_worker_tasks = self._run_workers(
+                "execute_model_parallel",
+                async_remote_only=True,
+                use_ray_compiled_dag=USE_RAY_COMPILED_DAG)
+            self.parallel_worker_tasks = asyncio.gather(*parallel_worker_tasks)
 
         # Only the driver worker returns the sampling results.
-        output = all_outputs[0]
-        return output
+        return self.driver_worker.execute_model(
+            seq_group_metadata_list=seq_group_metadata_list,
+            blocks_to_swap_in=blocks_to_swap_in,
+            blocks_to_swap_out=blocks_to_swap_out,
+            blocks_to_copy=blocks_to_copy)
+
+    def halt_model(self) -> None:
```

Comment:
Possibly a more straight-forward name:
```suggestion
    def stop_remote_worker_execution_loop(self) -> None:
```

Reply:
I've made this change, but the reason for the original name is that it's defined in the abstract base class where there is no concept of a remote worker. "halt_model" seemed like a more abstract way of describing it.